src/profile/notification-settings/utils.ts

```diff
@@ -0,0 +1,62 @@
+import { refreshSamUserAttributes, refreshTerraProfile } from 'src/auth/auth';
+import { Ajax } from 'src/libs/ajax';
+import Events, { EventWorkspaceAttributes, extractWorkspaceDetails } from 'src/libs/events';
+
+export const workspaceSubmissionNotificationKeys = (namespace: string, name: string) => [
+  `notifications/SuccessfulSubmissionNotification/${namespace}/${name}`,
+  `notifications/FailedSubmissionNotification/${namespace}/${name}`,
+  `notifications/AbortedSubmissionNotification/${namespace}/${name}`,
+];
+
+export const workspaceChangedNotificationKey = (namespace: string, name: string) => [
+  `notifications/WorkspaceChangedNotification/${namespace}/${name}`,
+];
+
+export const workspaceSubmissionNotificationInfo =
+  'Receive email notifications when a submission in this workspace has succeeded, failed, or been aborted.';
+export const workspaceChangedNotificationInfo =
+  'Receive email notifications when an owner of a workspace sends a workspace changed notification.';
+
+export type NotificationType =
+  | 'WorkspaceSubmission'
+  | 'WorkspaceChanged'
+  | 'GroupAccessRequest'
+  | 'WorkspaceAccessAdded'
+  | 'WorkspaceAccessRemoved'
+  | 'PlatformOperations'
+  | 'Marketing';
+
+const notificationKeysWithValue = (notificationKeys: string[], value: boolean): Record<string, string> => {
+  return Object.fromEntries(notificationKeys.map((notificationKey) => [notificationKey, JSON.stringify(value)]));
```

Comment:
Helper methods were from code originally in `NotificationSettings`, but I converted to TypeScript and removed usages of _ (after writing unit tests to capture the original behavior).